transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessengerTransport.kt

```diff
@@ -16,7 +16,17 @@ private const val TOKEN_STORE_KEY = "com.genesys.cloud.messenger"
 /**
  * The entry point to the services provided by the transport SDK.
  */
-class MessengerTransport(private val configuration: Configuration, private val tokenStore: TokenStore) {
+class MessengerTransport(
+    private val configuration: Configuration,
+    private val tokenStore: TokenStore,
+) {
+
+    private var messageClientImpl: MessagingClientImpl? = null
```

Comment:
That is a really good point here! 
There are a couple of things to consider: 

1.  I do not expect DeploymentConfig to be changed frequently. And even when they changed it might take some time for the BE to update with new configurations. Given that, I think it is reasonable to request from Transport consumer to restart MessagingClient if he/she wants to get the fresh DeploymentConfig. 
 
2. Regarding managing multiple MessagingClient instances:  There is a backend limitation of having more then 3 sessions opened with the same token. Any new MessagingClient instance calling `connect` will open a new session. The oldest one will be closed. While typing that, I realized that we have to capture that in our documentation. I have created a ticket for that: https://inindca.atlassian.net/browse/MTSDK-106

3. When we use a reference back approach on iOS, there is a risk of creating a retention cycle. We will have to be able to manage the subscribe/unsubscribe for each and every MessagingClient instance is created which have its own complications. 

Initial idea of updating of `MessangingClient.deploymentConfig` when `MessengerTransport.fetchDeploymentConfig()`  executed was to "save" on additional network request if it was already executed and not to keep the MessagingClient updated with latest data, as it is less frequent scenario.

Reply:
That's a good point on the `MessagingClient` limitation, I didn't realize that. Good idea to document the limitation.

Perhaps another option to consider is to discreetly keep and allow one instance of `MessagingClient` in `MessengerTransport`; an effective singleton. It's not like the caller can specify different `Configuration` objects to, say, connect to different servers. They're all created with the same `Configuration`. There's no use-case that I can think of that merits or even really supports multiple. 

Sure, it would be cleaner to then change the method to something like `getMessagingClient()` but we've already had a breaking API change there 😢.

Reply:
With regards to documenting the 3 session limitation, is that already documented in Web Messaging somewhere? Since that limitation isn't specific to the SDK it would be nice if we could just reference documentation for that behavior in Web Messaging instead of writing our own documentation that could end up going stale if there's a change in the backend.

Reply:
Note that if we do go with the "effective singleton" it's a moot point... there won't be a way to instantiate more than 1.

Reply:
After chewing on this a little more, I keep returning to the idea Brian first raised about having `MessengerTransport` be the source of truth for `DeploymentConfig`. Any clients generated by the `MessengerTransport` could have a reference back to it. As long as the references are only in one direction, the clients referencing back to the MessengerTransport that created them we shouldn't introduce any retain cycles.

Reply:
@crumpf @bjdupuis updated MessengerTransport to be a single source of truth for DeploymentConfig.